// radium-grid/lib/components/cell.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _react = require("react");

var _react2 = _interopRequireDefault(_react);

var _radium = require("radium");

var _radium2 = _interopRequireDefault(_radium);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

/* eslint-disable new-cap */
var Cell = function Cell(props) {
  return _react2.default.createElement(
    "div",
    { style: props.style },
    props.children
  );
};

var horizontalPropType = _react.PropTypes.oneOf(["left", "center", "right"]);
var verticalPropType = _react.PropTypes.oneOf(["top", "middle", "bottom"]);
Cell.propTypes = {
  width: _react.PropTypes.string,
  horizontalAlign: horizontalPropType,
  verticalAlign: verticalPropType,
  order: _react.PropTypes.number,

  smallWidth: _react.PropTypes.string,
  smallHorizontalAlign: horizontalPropType,
  smallVerticalAlign: verticalPropType,
  smallOrder: _react.PropTypes.number,

  mediumWidth: _react.PropTypes.string,
  mediumHorizontalAlign: horizontalPropType,
  mediumVerticalAlign: verticalPropType,
  mediumOrder: _react.PropTypes.number,

  largeWidth: _react.PropTypes.string,
  largeHorizontalAlign: horizontalPropType,
  largeVerticalAlign: verticalPropType,
  largeOrder: _react.PropTypes.number,

  xlargeWidth: _react.PropTypes.string,
  xlargeHorizontalAlign: horizontalPropType,
  xlargeVerticalAlign: verticalPropType,
  xlargeOrder: _react.PropTypes.number,

  children: _react.PropTypes.node,
  style: _react.PropTypes.object
};

exports.default = (0, _radium2.default)(Cell);
