// radium-grid/lib/components/grid.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; /* eslint-disable new-cap */


var _react = require("react");

var _react2 = _interopRequireDefault(_react);

var _radium = require("radium");

var _radium2 = _interopRequireDefault(_radium);

var _resolveCells = require("./util/resolve-cells");

var _resolveCells2 = _interopRequireDefault(_resolveCells);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var Grid = function Grid(props) {
  var styles = _extends({
    display: "flex",
    flexDirection: "row",
    flexWrap: "wrap",
    justifyContent: "space-between",
    minWidth: "100%"
  }, props.style);

  return _react2.default.createElement(
    "div",
    { style: styles },
    (0, _resolveCells2.default)(props)
  );
};

Grid.propTypes = {
  cellWidth: _react.PropTypes.string,
  cellAlign: _react.PropTypes.string,
  cellVerticalAlign: _react.PropTypes.string,

  smallCellWidth: _react.PropTypes.string,
  smallCellAlign: _react.PropTypes.string,
  smallCellVerticalAlign: _react.PropTypes.string,

  mediumCellWidth: _react.PropTypes.string,
  mediumCellAlign: _react.PropTypes.string,
  mediumCellVerticalAlign: _react.PropTypes.string,

  largeCellWidth: _react.PropTypes.string,
  largeCellAlign: _react.PropTypes.string,
  largeCellVerticalAlign: _react.PropTypes.string,

  xlargeCellWidth: _react.PropTypes.string,
  xlargeCellAlign: _react.PropTypes.string,
  xlargeCellVerticalAlign: _react.PropTypes.string,

  breakpoints: _react.PropTypes.shape({
    small: _react.PropTypes.string,
    medium: _react.PropTypes.string,
    large: _react.PropTypes.string,
    xlarge: _react.PropTypes.string
  }),

  gutter: _react.PropTypes.string,

  style: _react.PropTypes.object,
  children: _react.PropTypes.node
};

Grid.defaultProps = {
  cellWidth: "1/3",
  cellAlign: "left",
  cellVerticalAlign: "top",

  breakpoints: {
    small: "@media only screen and (max-width: 640px)",
    medium: "@media only screen and (min-width: 641px) and (max-width: 1024px)",
    large: "@media only screen and (min-width: 1025px) and (max-width: 1440px)",
    xlarge: "@media only screen and (min-width: 1441px)"
  },

  gutter: "16px"
};

exports.default = (0, _radium2.default)(Grid);
