// rebass/dist/Bar.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.styles = undefined;

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _createComponent = require('./create-component');

var _createComponent2 = _interopRequireDefault(_createComponent);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var styles = exports.styles = function styles(_ref, _ref2) {
  var scale = _ref.scale,
      bold = _ref.bold,
      colors = _ref.colors;
  var _ref2$value = _ref2.value,
      value = _ref2$value === undefined ? 0 : _ref2$value,
      backgroundColor = _ref2.backgroundColor;

  var p = value * 100 + '%';
  var backgroundImage = 'linear-gradient(90deg, ' + (backgroundColor || colors.blue) + ' ' + p + '%, transparent ' + p + ')';

  return {
    fontWeight: bold,
    padding: scale[1],
    backgroundSize: '100% 100%',
    backgroundImage: backgroundImage,
    backgroundColor: 'transparent'
  };
};

var Bar = (0, _createComponent2.default)('div', styles, {
  name: 'Bar',
  removeProps: ['value', 'color']
});

exports.default = Bar;
