// rebass/dist/Carousel.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.styles = undefined;

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _createComponent = require('./create-component');

var _createComponent2 = _interopRequireDefault(_createComponent);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _objectWithoutProperties(obj, keys) { var target = {}; for (var i in obj) { if (keys.indexOf(i) >= 0) continue; if (!Object.prototype.hasOwnProperty.call(obj, i)) continue; target[i] = obj[i]; } return target; }

/**
 * Carousel that wraps children and displays one at a time
 */

var styles = exports.styles = {
  root: {
    overflow: 'hidden'
  },
  inner: function inner(theme, _ref) {
    var index = _ref.index,
        length = _ref.length;
    return {
      whiteSpace: 'nowrap',
      height: '100%',
      transition: 'transform .2s ease-out',
      transform: 'translateX(' + index % length * -100 + '%)'
    };
  },
  slide: {
    display: 'inline-block',
    verticalAlign: 'middle',
    whiteSpace: 'normal',
    width: '100%',
    height: '100%'
  }
};

var Slide = (0, _createComponent2.default)('div', styles.slide);

var Inner = (0, _createComponent2.default)('div', styles.inner);

var Base = function Base(_ref2) {
  var _ref2$index = _ref2.index,
      index = _ref2$index === undefined ? 0 : _ref2$index,
      props = _objectWithoutProperties(_ref2, ['index']);

  var length = props.children ? props.children.length || 0 : 0;

  var children = _react2.default.Children.map(props.children, function (child, i) {
    return _react2.default.createElement(Slide, { children: child });
  });

  return _react2.default.createElement(
    'div',
    props,
    _react2.default.createElement(
      Inner,
      { index: index, length: length },
      children
    )
  );
};

var Carousel = (0, _createComponent2.default)(Base, styles.root, {
  name: 'Carousel'
});

exports.default = Carousel;
