// rebass/dist/List.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.styles = undefined;

var _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; };

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _createComponent = require('./create-component');

var _createComponent2 = _interopRequireDefault(_createComponent);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _objectWithoutProperties(obj, keys) { var target = {}; for (var i in obj) { if (keys.indexOf(i) >= 0) continue; if (!Object.prototype.hasOwnProperty.call(obj, i)) continue; target[i] = obj[i]; } return target; }

/** List component that accepts any child component item types */

var styles = exports.styles = function styles(_ref, _ref2) {
  var scale = _ref.scale;
  var reset = _ref2.reset,
      flush = _ref2.flush;
  return {
    margin: 0,
    paddingLeft: flush || reset ? 0 : scale[2],
    listStyle: reset ? 'none' : null
  };
};

var wrapChildren = function wrapChildren(children) {
  return _react2.default.Children.map(children, function (child) {
    return child.type === 'li' ? child : _react2.default.createElement(
      'li',
      null,
      child
    );
  });
};

var Base = function Base(_ref3) {
  var ordered = _ref3.ordered,
      props = _objectWithoutProperties(_ref3, ['ordered']);

  var Comp = ordered ? 'ol' : 'ul';
  var children = wrapChildren(props.children);

  return _react2.default.createElement(Comp, _extends({}, props, {
    children: children
  }));
};

var List = (0, _createComponent2.default)(Base, styles, {
  name: 'List'
});

exports.default = List;
