// rebass/dist/SignInForm.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; };

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _classnames = require('classnames');

var _classnames2 = _interopRequireDefault(_classnames);

var _withRebass = require('./withRebass');

var _withRebass2 = _interopRequireDefault(_withRebass);

var _Heading = require('./Heading');

var _Heading2 = _interopRequireDefault(_Heading);

var _Input = require('./Input');

var _Input2 = _interopRequireDefault(_Input);

var _Checkbox = require('./Checkbox');

var _Checkbox2 = _interopRequireDefault(_Checkbox);

var _Button = require('./Button');

var _Button2 = _interopRequireDefault(_Button);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _objectWithoutProperties(obj, keys) { var target = {}; for (var i in obj) { if (keys.indexOf(i) >= 0) continue; if (!Object.prototype.hasOwnProperty.call(obj, i)) continue; target[i] = obj[i]; } return target; }

var SignInForm = function SignInForm(_ref) {
  var _ref$heading = _ref.heading,
      heading = _ref$heading === undefined ? 'Sign In' : _ref$heading,
      _ref$username = _ref.username,
      username = _ref$username === undefined ? '' : _ref$username,
      _ref$password = _ref.password,
      password = _ref$password === undefined ? '' : _ref$password,
      _ref$remember = _ref.remember,
      remember = _ref$remember === undefined ? false : _ref$remember,
      _ref$onChange = _ref.onChange,
      onChange = _ref$onChange === undefined ? function () {} : _ref$onChange,
      _ref$errors = _ref.errors,
      errors = _ref$errors === undefined ? {} : _ref$errors,
      className = _ref.className,
      style = _ref.style,
      theme = _ref.theme,
      subComponentStyles = _ref.subComponentStyles,
      props = _objectWithoutProperties(_ref, ['heading', 'username', 'password', 'remember', 'onChange', 'errors', 'className', 'style', 'theme', 'subComponentStyles']);

  // const {} = theme

  var cx = (0, _classnames2.default)('SignInForm', className);

  var sx = {
    root: _extends({}, style)
  };

  // Or Create an Account...

  return _react2.default.createElement(
    'form',
    _extends({}, props, {
      className: cx,
      style: sx.root }),
    heading && _react2.default.createElement(_Heading2.default, { mb: 2, children: heading }),
    _react2.default.createElement(_Input2.default, {
      name: 'username',
      label: 'Username',
      value: username,
      onChange: onChange,
      message: errors.username
    }),
    _react2.default.createElement(_Input2.default, {
      type: 'password',
      name: 'password',
      label: 'Password',
      value: password,
      onChange: onChange,
      message: errors.password
    }),
    _react2.default.createElement(_Checkbox2.default, {
      name: 'remember',
      label: 'Remember me on this device',
      mb: 2,
      checked: remember,
      onClick: onChange
    }),
    _react2.default.createElement(_Button2.default, { children: 'Sign In' }),
    _react2.default.createElement(
      'a',
      { href: '#' },
      'Forgot password?'
    )
  );
};

SignInForm._name = 'SignInForm';

exports.default = (0, _withRebass2.default)(SignInForm);
