// rebass/dist/TabItem.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.styles = undefined;

var _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; };

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _createComponent = require('./create-component');

var _createComponent2 = _interopRequireDefault(_createComponent);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var styles = exports.styles = function styles(_ref, _ref2) {
  var typeScale = _ref.typeScale,
      bold = _ref.bold,
      scale = _ref.scale,
      colors = _ref.colors;
  var active = _ref2.active;
  return _extends({
    fontSize: typeScale[5],
    fontWeight: bold,
    paddingTop: scale[1],
    paddingBottom: scale[1],
    marginRight: scale[2],
    textDecoration: 'none',
    borderBottom: '2px solid transparent',
    cursor: 'pointer'
  }, active ? {
    color: colors.blue,
    borderColor: colors.blue
  } : {});
};

var TabItem = (0, _createComponent2.default)('a', styles, {
  name: 'TabItem',
  removeProps: ['active']
});

exports.default = TabItem;
