// rebass/dist/themes/basic.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; };

var _understyle = require('understyle');

var colors = _understyle.defaultConfig.colors;


var serif = 'Georgia, serif';
var monospace = 'Menlo, Consolas, monospace';

var scale = [0, 8, 16, 32, 64];

var typeScale = [48, 32, 24, 20, 16, 14, 12];

var zIndex = [0, 2, 4, 8, 16];

var bold = 600;
var borderRadius = 2;
var borderColor = 'rgba(0, 0, 0, .25)';
var boxShadow = '0 2px 16px rgba(0, 0, 0, .125)';

var caps = {
  textTransform: 'uppercase',
  letterSpacing: '.2em'
};

var basic = {
  name: 'Basic',
  scale: scale,
  typeScale: typeScale,
  bold: bold,
  serif: serif,
  monospace: monospace,
  caps: caps,
  zIndex: zIndex,
  colors: colors,
  borderRadius: borderRadius,
  borderColor: borderColor,
  boxShadow: boxShadow,
  Heading: {
    alt: _extends({}, caps)
  }
};

exports.default = basic;
